li-utils/src/main/javaPegasus/com/linkedin/common/urn/UrnValidator.java

```diff
@@ -0,0 +1,30 @@
+package com.linkedin.common.urn;
+
+import com.linkedin.data.message.Message;
+import com.linkedin.data.schema.DataSchema;
+import com.linkedin.data.schema.NamedDataSchema;
+import com.linkedin.data.schema.validator.Validator;
+import com.linkedin.data.schema.validator.ValidatorContext;
+import java.net.URISyntaxException;
+
+
+/**
+ * Rest.li Validator responsible for ensuring that {@link Urn} objects are well-formed.
+ *
+ * Note that this validator does not validate the integrity of strongly typed urns,
+ * or validate Urn objects against their associated key aspect.
+ */
+public class UrnValidator implements Validator {
+  @Override
+  public void validate(ValidatorContext context) {
+    if (DataSchema.Type.TYPEREF.equals(context.dataElement().getSchema().getType())
```

Comment:
should we also try to construct the key here to verify the correct number & types of params are present?

Reply:
+1

Reply:
As discussed offline, this layer is purely for structural (schema-based) validation. There's no way to know what entity key to validate against at this context

Reply:
Nonetheless, we should add urn->key checks in EntityService (will do separately)